docs/calendar.md

```diff
@@ -103,6 +103,38 @@ Custom calendars should maintain this behavior unless there's a good reason not
+
+Calendars may also have one-time changes.
+For example, 10-14 days were skipped when Western countries switched from the [Julian calendar](https://en.wikipedia.org/wiki/Julian_calendar) to the [Gregorian calendar](https://en.wikipedia.org/wiki/Gregorian_calendar).
+In France, 4 October 1582 (the last day of the Julian calendar) was followed by 15 October 1582, skipping 10 calendar days.
+The built-in `gregory` calendar in ECMAScript is a [proleptic Gregorian calendar](https://en.wikipedia.org/wiki/Proleptic_Gregorian_calendar); it ignores the skipped days between Julian and Gregorian dates by projecting Gregorian calendar rules backwards in time.
+Other calendars, either future built-in calendars or custom calendars, may skip days, months, or even years.
+
+Calendar variation across years means that programs may encounter historical dates that are valid in one year but invalid in another.
+A common example is calling `toPlainDate` on a `Temporal.PlainMonthDay` object to convert a month/day pair like a birthday or anniversary (which could have been on a leap day, leap month, or other skipped period) into a `Temporal.PlainDate`.
+Temporal types' `with` or `from` methods can run into the same issue.
+When Temporal encounters inputs representing a month and day that doesn't exist in the desired calendar year, by default (which can be overridden in `with` or `from` via the `overflow` option) the provided fields will be adjusted using the following algorithm:
+* First, pick the closest `day` in the same month. If there are two equally-close dates in that month, pick the later one.
+* If the month is a leap month that doesn't exist in the desired year, then pick another date according to the cultural conventions of that calendar's users. Usually this will result in the same `day` in the month before or the month after where that month would normally fall in a leap year.
+* Otherwise, pick the closest date to the provided date that is still in the same year. If there are two equally-close dates, pick the later one.
+* If the entire year doesn't exist, then pick the closest date to the provided date. If there are two equally-close dates, pick the later one.
+
+Because of the rules above, the following invariants (statements that are always true) will hold for any `Temporal.PlainDate`, `Temporal.PlainDateTime`, or `Temporal.ZonedDateTime` object:
+* `obj.with({ day: 1 })` will always return the first day of the object's `month`, even if the resulting `day` is not 1 because earlier days were skipped.
+* `obj.with({ day: Infinity })` will always return the last day of the object's `month`.
+* `obj.with({ month: 1, day: 1 })` will always return the first day of the object's `year`. `day` is not guaranteed to be 1, but `month` will always be 1 because the `month` property of calendars always refers to the order of months in that year.
+* `obj.with({ month: Infinity, day: Infinity })` will always return the last day of the object's `year`.
```

Comment:
Likewise, `{ month: obj.monthsInYear, day: obj.daysInYear }`? (`monthsInYear` _does_ work here because `month` is a 1-based index)